torchtune/training/checkpointing/_checkpointer.py

```diff
@@ -537,8 +555,28 @@ def save_checkpoint(
                     tie_word_embeddings=self._config["tie_word_embeddings"],
                 )
             elif self._model_type == ModelType.LLAMA3_VISION:
-                raise NotImplementedError(
-                    "Safe tensors support for LLaMA3 Vision will be added soon. Pleaee use the FullModelMetaCheckpointer for now."
+                from torchtune.models.llama3_2_vision._convert_weights import (
+                    llama3_vision_tune_to_hf,
+                )
+
+                text_config = self._config.get("text_config", {})
+                vision_config = self._config.get("vision_config", {})
+                state_dict[training.MODEL_KEY] = llama3_vision_tune_to_hf(
+                    state_dict[training.MODEL_KEY],
+                    num_heads=text_config["num_attention_heads"],
+                    num_kv_heads=text_config["num_key_value_heads"],
+                    dim=text_config["hidden_size"],
+                    head_dim=text_config.get("head_dim", None),
```

Comment:
why do some of these have defaults for `.get` and some do not have defaults? could `text_config` and `vision_config` ever be empty?

Reply:
I don't know, I was following the base implementation and all lists defaulted to None.